### PR TITLE
Move issue status to milestones from labels.

### PR DESCRIPTION
**Pull request by miketaylr**
Tests pass, and this works on staging.

@karlcow is there anything else we need to do before pulling this trigger and deploying to production? Your email about needsinfo comes to mind.

### CODE REVIEW

File: config/__init__.py [hunk 1]
```diff
@@ -4,12 +4,67 @@
...
+        milestones_content = r.content
+        if r.status_code == 200:
+            with open(milestones_path, 'w') as f:
+                f.write(r.content)
+            print('Milestones saved in data/')
+        r.raise_for_status()
+    except requests.exceptions.HTTPError as error:
+        # Not working, let's use the cached copy
+        # This might fail the first time.
+        print(MILESTONE_ERROR.format(msg=error))
```
Comment (miketaylr) [inline config/__init__.py]: Ah, this will probably throw. `MILESTONE_ERROR` is defined in run.py. We should fix this before merging.

Comment (miketaylr) [reply config/__init__.py]: OK done.

**Comment by miketaylr** [comment]
@magsout there's a lot of code in here, but would you mind giving it a review for sanity check?

**Comment by magsout** [comment]
@miketaylr sure 👍

File: webcompat/templates/issue/issue-labels.jst [hunk 2]
```diff
@@ -1,16 +1,16 @@
...
-    <div class="wc-Issue-labelEditor-title">
+  <header class="wc-Issue-categoryEditor">
+    <div class="wc-Issue-categoryEditor-title">
       Labels
     </div>
     <% if ($("body").data("username")) { %>
-    <span class="wc-Issue-labelEditor-button js-Issue-labelEditor">
-      <button class="wc-LabelEditorLauncher r-ResetButton wc-Icon wc-Icon--gear js-LabelEditorLauncher" aria-hidden="true">
+    <span class="wc-Issue-categoryEditor-button js-Issue-categoryEditor">
+      <button class="wc-CategoryEditorLauncher r-ResetButton wc-Icon wc-Icon--gear js-CategoryEditorLauncher js-LabelEditorLauncher" aria-hidden="true">
```
Comment (magsout) [inline webcompat/templates/issue/issue-labels.jst]: Why not delete this class `js-LabelEditorLauncher` ? If I understand, you renamed `LabelEditor` by `CategoryEditor`

File: webcompat/static/js/lib/labels.js [hunk 3]
```diff
@@ -15,42 +15,19 @@ if (!issues.allLabels) {
...
-  },
   closeEditor: function() {
     this.labelEditor.closeEditor();
   },
-  renderLabels: function() {
-    this.$el.html(this.template(this.model.toJSON()));
-  },
-  fetchLabels: function() {
+  fetchItems: function() {
     this.editorButton = $(".js-LabelEditorLauncher");
```
Comment (magsout) [inline webcompat/static/js/lib/labels.js]: This class should be updated by `js-CategoryEditorLauncher`?

File: webcompat/static/js/lib/labels.js [hunk 4]
```diff
@@ -64,7 +41,13 @@ issues.LabelsView = Backbone.View.extend({
...
+  openEditor: function(e) {
+    // make sure we're not typing in the comment textfield.
+    if (e && e.target.nodeName === "TEXTAREA") {
+      return;
+    }
+
+    this.labelEditor.isOpen = true;
     this.editorButton.addClass("is-active");
     this.$el
       .find(".js-LabelEditorLauncher")
```
Comment (magsout) [inline webcompat/static/js/lib/labels.js]: Same here that my comment above

File: webcompat/static/js/lib/issues.js [hunk 5]
```diff
@@ -398,28 +314,43 @@ issues.MainView = Backbone.View.extend({
...
-        target.hasClass("js-LabelEditorLauncher")) ||
+        target.hasClass("js-CategoryEditorLauncher")) ||
       // or clicked anywhere inside the label editor
-      target.parents(".js-LabelEditor").length
+      target.parents(".js-CategoryEditor").length
     ) {
+      // Clicking on one launcher will force to close the other one
+      if (
+        target[0].nodeName === "BUTTON" &&
+        target.hasClass("js-LabelEditorLauncher")
```
Comment (magsout) [inline webcompat/static/js/lib/issues.js]: We should updated this class by `js-CategoryEditorLauncher`

File: tests/functional/labels-auth.js [hunk 6]
```diff
@@ -39,7 +39,7 @@ define(
             .findByCssSelector(".js-LabelEditorLauncher")
```
Comment (magsout) [inline tests/functional/labels-auth.js]: we should updated `.js-LabelEditorLauncher` by `.js-CategoryEditorLauncher` ?

**Comment by magsout** [comment]
Very good job @MDTsai @karlcow @miketaylr 👍

**Comment by miketaylr** [comment]
> My only comment is about labelEditor and CategoryEditor, If I understand well you renamed LabelEditor by CategoryEditor.

Sort of... what we did was create a more generic CategoryEditor that labelEditor and milestonesEditor can inherit from. 

So we need a way to be able to identify each editor from eachother, that's why we have `js-MilestoneEditorLauncher` as well. Does that make sense?

**Comment by magsout** [comment]
@miketaylr yeah, ok, make sense.

**Comment by miketaylr** [comment]
@karlcow any updates on if this is safe to land/deploy now?

**Comment by karlcow** [comment]
@miketaylr in the process https://github.com/webcompat/devops/issues/18 The script will be finished this week.

**Comment by miketaylr** [comment]
Time to merge.

**Comment by zoepage** [comment]
@miketaylr is it safe to delete this branch? :)

**Comment by miketaylr** [comment]
> @miketaylr is it safe to delete this branch? :)

yes!

**Pull request merged**